File: backend/app/schemas/jobs.py
from pydantic import BaseModel, HttpUrl

from app.models.job import ClipAsset, JobStatus


class CreateJobRequest(BaseModel):
    youtube_url: HttpUrl


class RegenerateClipRequest(BaseModel):
    clip_id: str


class ClipResponse(ClipAsset):
    pass


class JobResponse(BaseModel):
    job_id: str
    source_url: HttpUrl
    status: JobStatus
    progress: int
    error: str | None
    clips: list[ClipResponse]

File: frontend/src/types/jobs.ts
export type Clip = {
  clip_id: string
  title: string
  summary: string
  start_seconds: number
  end_seconds: number
  duration_seconds: number
  transcript_excerpt: string
  file_path?: string | null
  subtitles_path?: string | null
}

export type Job = {
  job_id: string
  source_url: string
  status: 'queued' | 'processing' | 'completed' | 'failed'
  progress: number
  error?: string | null
  clips: Clip[]
}
